main/src/main/java/org/mobilitydata/gtfsvalidator/table/GeoJsonFileLoader.java

```diff
@@ -0,0 +1,224 @@
+package org.mobilitydata.gtfsvalidator.table;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import org.locationtech.jts.geom.*;
+import org.mobilitydata.gtfsvalidator.notice.*;
+import org.mobilitydata.gtfsvalidator.util.geojson.GeoJsonGeometryValidator;
+import org.mobilitydata.gtfsvalidator.util.geojson.GeometryType;
+import org.mobilitydata.gtfsvalidator.util.geojson.UnparsableGeoJsonFeatureException;
+import org.mobilitydata.gtfsvalidator.validator.ValidatorProvider;
+
+/**
+ * This class knows how to load a GeoJSON file. Typical GeoJSON file: { "type": "FeatureCollection",
+ * "features": [ { "id": "area_548", "type": "Feature", "geometry": { "type": "Polygon",
+ * "coordinates": [ [ [ -122.4112929, 48.0834848 ], ... ] ] }, "properties": { "stop_name": "Some
+ * name", "stop_desc": "Some description" } }, ... ] }
+ */
+public class GeoJsonFileLoader extends TableLoader {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private GeoJsonGeometryValidator geometryValidator;
+
+  @Override
+  public GtfsEntityContainer load(
+      GtfsFileDescriptor fileDescriptor,
+      ValidatorProvider validatorProvider,
+      InputStream inputStream,
+      NoticeContainer noticeContainer) {
+    GtfsGeoJsonFileDescriptor geoJsonFileDescriptor = (GtfsGeoJsonFileDescriptor) fileDescriptor;
+    geometryValidator = new GeoJsonGeometryValidator(noticeContainer);
+    try {
+      List<GtfsGeoJsonFeature> entities = extractFeaturesFromStream(inputStream, noticeContainer);
+      return geoJsonFileDescriptor.createContainerForEntities(entities, noticeContainer);
+    } catch (JsonParseException jpex) {
+      noticeContainer.addValidationNotice(new MalformedJsonNotice(GtfsGeoJsonFeature.FILENAME));
+      logger.atSevere().withCause(jpex).log("Malformed JSON in locations.geojson");
+      return fileDescriptor.createContainerForInvalidStatus(TableStatus.UNPARSABLE_ROWS);
+    } catch (IOException ioex) {
+      noticeContainer.addSystemError(new IOError(ioex));
+      return fileDescriptor.createContainerForInvalidStatus(TableStatus.UNPARSABLE_ROWS);
+    } catch (UnparsableGeoJsonFeatureException ugex) {
+      logger.atSevere().withCause(ugex).log("Unparsable GeoJSON feature");
```

Comment:
Is there a reason why we generate a notice for JsonParseException but not for UnparsableGeoJsonFeatureException?